src/Gui/DlgSettingsColorGradientImp.cpp

```diff
@@ -60,7 +60,10 @@ DlgSettingsColorGradientImp::DlgSettingsColorGradientImp( QWidget* parent, Qt::W
  */
 DlgSettingsColorGradientImp::~DlgSettingsColorGradientImp()
 {
-    // no need to delete child widgets, Qt does it all for us
+    // no need to delete child widgets, Qt does it all for us
+    // Validators are not owned by the widgets
+    delete fMaxVal;
+    delete fMinVal;

```

Comment:
Explicitly deleting the validators here do no harm but is superfluous because contrary to your comment they are owned by the dialog:
```
    fMaxVal = new QDoubleValidator(-1000,1000,ui->spinBoxDecimals->maximum(),this);
    ui->floatLineEditMax->setValidator(fMaxVal);
    fMinVal = new QDoubleValidator(-1000,1000,ui->spinBoxDecimals->maximum(),this);
    ui->floatLineEditMin->setValidator(fMinVal);
```

The third argument in the constructor is the parent. So, when the parent (i.e. this dialog) is deleted it should also delete its children.

Reply:
You are absolutely right on that one. I missed the parent That was a good start  for me :D :D

Reply:
> It technically DOES hurt, in theory this may lead to releasing something irrelavant.

No, it (usually) doesn't hurt. When a sub-object is explicitly deleted before its parent then it gets a notification be Qt and makes sure to avoid a double destruction.

Anyway, when merging this PR I will omit this change...